schemaReporter pollTimer has incorrect type

### Issue Description

In packages/server/src/plugin/schemaReporting/schemaReporter.ts

pollTimer is typed as NodeJS.Timer, it is actually a timeout, so using node 18> types it is incompatible
with clearTimeout()

It should be:

`private pollTimer?: ReturnType<typeof setTimeout>;`


```
node_modules/@apollo/server/src/plugin/schemaReporting/schemaReporter.ts:90:20 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
    Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
      Property '[Symbol.dispose]' is missing in type 'Timer' but required in type 'Timeout'.
  Overload 2 of 2, '(id: number | undefined): void', gave the following error.
    Argument of type 'Timer' is not assignable to parameter of type 'number'.

90       clearTimeout(this.pollTimer);
                      ~~~~~~~~~~~~~~

  node_modules/@types/node/timers.d.ts:130:17
    130                 [Symbol.dispose](): void;
                        ~~~~~~~~~~~~~~~~
    '[Symbol.dispose]' is declared here.

```

### Link to Reproduction

N/A

### Reproduction Steps

Upgrade your node types to 18, but it is pretty obvious that the type is incorrect.

--- a/.changeset/late-donuts-think.md
+++ b/.changeset/late-donuts-think.md
@@ -1,0 +1,5 @@
+---
+'@apollo/server': patch
+---
+
+Change SchemaReporter.pollTimer from being a NodeJS.Timer to a NodeJS.Timeout

--- a/packages/server/src/plugin/schemaReporting/schemaReporter.ts
+++ b/packages/server/src/plugin/schemaReporting/schemaReporter.ts
@@ -40,7 +40,7 @@ export class SchemaReporter {
   private readonly fetcher: Fetcher;
 
   private isStopped: boolean;
-  private pollTimer?: NodeJS.Timer;
+  private pollTimer?: ReturnType<typeof setTimeout>;
   private readonly headers: Record<string, string>;
 
   constructor(options: {

--- a/packages/server/src/plugin/schemaReporting/schemaReporter.ts
+++ b/packages/server/src/plugin/schemaReporting/schemaReporter.ts
@@ -40,7 +40,7 @@ export class SchemaReporter {
   private readonly fetcher: Fetcher;
 
   private isStopped: boolean;
-  private pollTimer?: ReturnType<typeof setTimeout>;
+  private pollTimer?: NodeJS.Timeout;
   private readonly headers: Record<string, string>;
 
   constructor(options: {

